components/raftstore-v2/src/batch/store.rs

```diff
@@ -249,24 +289,44 @@ where
     EK: KvEngine,
     T: Transport + 'static,
 {
-    type Handler = StorePoller<EK, T>;
+    type Handler = StorePoller<EK, ER, T>;
 
     fn build(&mut self, priority: batch_system::Priority) -> Self::Handler {
         let poll_ctx = StoreContext::new(
             self.cfg.value().clone(),
             self.trans.clone(),
+            self.router.clone(),
+            self.write_senders.clone(),
             self.logger.clone(),
         );
         let cfg_tracker = self.cfg.clone().tracker("raftstore".to_string());
         StorePoller::new(poll_ctx, cfg_tracker)
     }
 }
 
+/// A set of background threads that will processing offloaded work from
+/// raftstore.
+struct Workers<EK: KvEngine, ER: RaftEngine> {
```

Comment:
In the future, shall we consider to use thread pool for these background workers, instead of dedicated thread for each worker?

Reply:
Thread pool is already used. But there are still some background work that can take too much time and starving all other jobs. So dedicated thread is used for those tasks.